src/main/java/spoon/reflect/visitor/DefaultJavaPrettyPrinter.java

```diff
@@ -400,7 +402,7 @@ public DefaultJavaPrettyPrinter scan(CtElement e) {
 	}
 
 	private boolean shouldSetBracket(CtExpression<?> e) {
-		if (!e.getTypeCasts().isEmpty()) {
+		if (!e.getTypeCasts().isEmpty() && !isMinimizeRoundBrackets()) {
```

Comment:
This additional check can still cause issues, because now we can go through the whole method and return `false` as a default value, even if there *are* type casts. It might work by returning `!e.getTypeCasts().isEmpty()` at the bottom,

Consider following silly example:
```java
class A {
    A a;
    int i;
}
class C extends A {
    Object a;
}
```
(the subclass shadows a field of its superclass)
and following statement:
```java
C c = null;
int i = ((A) c).a.i;
```
This currently does not enter the if body but instead return `false` at the end of the method. Obviously, `(A) c.a.i` does not compile.

Even worse, we can construct code that still compiles but has different semantics by adjusting the classes as follows:

```java
class A {
    A a;
    int i;
}
class C extends A {
    C a;
		A i;
}
```
and the statements:
```java
C c = null;
var i = ((A) c).a.i;
```